geocruncher/topography_reader.py

```diff
@@ -10,23 +10,17 @@
 import numpy as np
 from gmlib.topography_reader import ImplicitDTM
 
-
 def ascii_grid_to_implicit_dtm(dem: str) -> ImplicitDTM:
     """Read ASCIIGrid DEM datapoints and return a GMLIB ImplicitDTM."""
-    with StringIO(dem) as f:
-        # We do not actually care about the first 2 line (ncols, nrows), skip them
-        f.readline()
-        f.readline()
-        xllcorner = float(re.findall(r"-?\d+\.?\d+", f.readline())[0])
-        yllcorner = float(re.findall(r"-?\d+\.?\d+", f.readline())[0])
-        cellsize = float(re.findall(r"\d+\.?\d*", f.readline())[0])
-        f.readline()
-        zmap = []
-        line = f.readline().strip().split()
-        while line:
-            reversed_arr = np.array([float(s) for s in line])
-            zmap.append(reversed_arr)
-            line = f.readline().strip().split()
-        zmap = np.array(zmap[::-1])
+        
+    lines = dem.splitlines()
+    # We do not actually care about the first 2 line (ncols, nrows), skip them
+    xllcorner = float(re.search(r"-?\d+\.?\d*", lines[2])[0])
+    yllcorner = float(re.search(r"-?\d+\.?\d*", lines[3])[0])
+    cellsize = float(re.search(r"\d+\.?\d*", lines[4])[0])
+    
+    data_string = '\n'.join(lines[6:])
+    zmap = np.loadtxt(StringIO(data_string), dtype=np.float64)
```

Comment:
Does the `StringIO` get closed properly when used like this ?

Reply:
Yes, the StringIO object gets closed properly in this case. Since we're creating a temporary StringIO object that's passed directly to loadtxt, Python's garbage collector will automatically clean it up when it's no longer needed. StringIO is an in-memory object that doesn't require explicit resource management like physical files do.